src/proxy.rs

```diff
@@ -469,6 +477,98 @@ impl Dst for Uri {
     }
 }
 
+/// Get system proxies information.
+///
+/// It can only support Linux, Unix like, and windows system.  Note that it will always
+/// return a HashMap, even if something runs into error when find registry information in
+/// Windows system.  Note that invalid proxy url in the system setting will be ignored.
+///
+/// Returns:
+///     System proxies information as a hashmap like
+///     {"http": Url::parse("http://127.0.0.1:80"), "https": Url::parse("https://127.0.0.1:80")}
+pub fn get_proxies() -> HashMap<String, Url> {
+    let proxies: HashMap<String, Url> = get_from_environment();
+
+    if proxies.is_empty() {
+        // don't care errors if can't get proxies from registry, just return an empty HashMap.
+        #[cfg(target_os = "windows")]
```

Comment:
This `cfg` could go on the `if proxies.is_empty()` instead, and so the `if` would never be checked on *nix.

Reply:
Attribute for `if` expression doesn't support yet.  So I have to use the `cfg` inside `if`.  Here is [relative document](https://doc.rust-lang.org/reference/expressions.html#expression-attributes)

Reply:
Oh yes, that. There's a silly way to get around it: just put the whole thing in a block.

```rust
#[cfg(target_os = "windows")]
{
    if proxies.is_empty() {
        // etc
    }
}
```